addons/l10n_it_edi/tests/test_ir_mail_server.py

```diff
@@ -158,3 +180,140 @@ def test_ricevuta_consegna(self):
         """ Test a receipt adapted from https://www.fatturapa.gov.it/export/documenti/messaggi/v1.0/IT01234567890_11111_DT_001.xml """
         self._test_receipt('DT', 'delivered', 'delivered_expired')
+
+    def test_price_included_taxes(self):
+        """ When the tax is price included, there should be a rounding value added to the xml, if the sum(subtotals) * tax_rate is not
+            equal to taxable base * tax rate (there is a constraint in the edi where taxable base * tax rate = tax amount, but also
+            taxable base = sum(subtotals) + rounding amount)
+        """
+        price_included_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.italian_partner_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    'name': 'something price included',
+                    'quantity': 1,
+                    'price_unit': 800.40,
+                    'tax_ids': [(6, 0, [self.price_included_tax.id])]
+                }),
+                (0, 0, {
+                    'name': 'something else price included',
+                    'quantity': 1,
+                    'price_unit': 800.40,
+                    'tax_ids': [(6, 0, [self.price_included_tax.id])]
+                }),
+                (0, 0, {
+                    'name': 'something not price included',
+                    'quantity': 1,
+                    'price_unit': 800.40,
+                    'tax_ids': [(6, 0, [self.tax_sale_a.id])]
+                }),
+            ],
+        })
+
+        # In this case, the first two lines use a price_include tax the
+        # subtotals should be 800.40 / (100 + 22.0) * 100 = 656.065564..,
+        # where 22.0 is the tax rate.
+        #
+        # Since the subtotals are rounded we actually have 656.07
+        lines = price_included_invoice.line_ids
+        price_included_lines = lines.filtered(lambda line: line.tax_ids == self.price_included_tax)
+        self.assertEqual([line.price_subtotal for line in price_included_lines], [656.07, 656.07])
+        # So the taxable a base the edi expects (for this tax) is actually 1312.14
+        price_included_tax_line = lines.filtered(lambda line: line.tax_line_id == self.price_included_tax)
+        self.assertEqual(price_included_tax_line.tax_base_amount, 1312.14)
+
+        # The tax amount of the price included tax should be:
+        #   per line: 800.40 - (800.40 / (100 + 22) * 100) = 144.33
+        #   tax amount: 144.33 * 2 = 288.66
+        self.assertEqual(price_included_tax_line.price_total, 288.66)
+
+        # Prepare the XML
+        template_values = price_included_invoice._prepare_fatturapa_export_values()
+        xml = self.env.ref('l10n_it_edi.account_invoice_it_FatturaPA_export')._render(template_values)
+        tree = etree.fromstring(xml)
+
+        expected_xml = self._get_test_file_content('IT00470550013_price_included.xml')
```

Comment:
Now that test_invoices are more than two, we could use a dict for all the invoices that we are to test, (instead of single variables) like
```py
cls.test_invoices = {k: cls._get_test_file_content(v) for k, v in [
                        ('normal_1', 'IT01234567890_FPR01.xml'),
                        ('normal_2', 'IT01234567890_FPR02.xml'),
                        ('signed', 'IT01234567890_FPR01.xml.p7m'),
                        ('price_included', 'IT00470550013_price_included.xml'),
                        ('discount_mixed', 'IT00470550013_discounted_mixed.xml'),
                    ]}
```
so that you can 
```py
        expected_tree = etree.fromstring(self.test_invoices['price_included'])
```